### PR TITLE
Support flat container icon URL in Azure Search

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/NuGetGallery/issues/7261

Another PR will happen when @agr finished the DB schema for icons. For now, only Catalog2AzureSearch will populate icon URL for packages with icon files.

Similar to PR https://github.com/NuGet/NuGet.Services.Metadata/pull/556.

### CODE REVIEW

File: src/NuGet.Services.AzureSearch/DocumentUtilities.cs [hunk 1]
```diff
@@ -116,13 +117,19 @@ public static string GetSearchFilterString(SearchFilters searchFilters)
...
+            // https://github.com/NuGet/NuGetGallery/issues/7261
+            // https://github.com/nuget/nugetgallery/issues/7064
         }
 
         public static void PopulateMetadata(
             IBaseMetadataDocument document,
             string normalizedVersion,
             PackageDetailsCatalogLeaf leaf,
-            Uri galleryBaseUrl)
+            Uri galleryBaseUrl,
```
Comment (loic-sharma) [inline src/NuGet.Services.AzureSearch/DocumentUtilities.cs]: Consider making this class non-static and passing these configs through constructor. You can make `FlatContainerPackagePathProvider` a field that's created by dependency injection too.

Comment (joelverhagen) [reply src/NuGet.Services.AzureSearch/DocumentUtilities.cs]: Done. Left `FlatContainerPackagePathProvider` as a non-field to respect the option's reload semantics.

File: tests/NuGet.Services.AzureSearch.Tests/HijackDocumentBuilderFacts.cs [hunk 2]
```diff
@@ -395,12 +395,24 @@ public void SetsLicenseUrlToGalleryWhenPackageHasLicenseFile()
 
                 Assert.Equal(Data.GalleryLicenseUrl, document.LicenseUrl);
             }
+
+            [Fact]
+            public void SetsIconUrlToFlatContainerWhenPackageHasIconFile()
```
Comment (agr) [inline tests/NuGet.Services.AzureSearch.Tests/HijackDocumentBuilderFacts.cs]: This seem to only test the situation when we have both iconUrl and iconFile specified. How about other combinations?

Comment (joelverhagen) [reply tests/NuGet.Services.AzureSearch.Tests/HijackDocumentBuilderFacts.cs]: just `iconUrl` is already tested. I will add one for just `iconFile`

**Pull request merged**